pallets/xyk/src/lib.rs

```diff
@@ -429,17 +429,29 @@ impl<T: Trait> Module<T> {
         output_reserve: Balance,
         sell_amount: Balance,
     ) -> Result<Balance, DispatchError> {
-        let input_reserve_saturated: U256 = input_reserve.into();
-        let output_reserve_saturated: U256 = output_reserve.into();
-        let sell_amount_saturated: U256 = sell_amount.into();
+        let after_fee_percentage: u128 = 10000 - FEE_PERCENTAGE;
+        let input_reserve_saturated: U256 = input_reserve.saturated_into::<u128>().into();
+        let output_reserve_saturated: U256 = output_reserve.saturated_into::<u128>().into();
+        let sell_amount_saturated: U256 = sell_amount.saturated_into::<u128>().into();
 
-        let input_amount_with_fee: U256 = sell_amount_saturated * 997;
-        let numerator: U256 = input_amount_with_fee * output_reserve_saturated;
-        let denominator: U256 = input_reserve_saturated * 1000 + input_amount_with_fee;
-        let result = numerator
-            .checked_div(denominator)
+        let input_amount_with_fee: U256 = sell_amount_saturated
+            .checked_mul(after_fee_percentage.into())
```

Comment:
isn't inclusion of the checked math in conflict with https://github.com/mangata-finance/mangata-node/pull/63 ?